xcdat/regridder/xesmf.py

```diff
@@ -0,0 +1,192 @@
+        extrap_dist_exponent: float = None,
+        extrap_num_src_pnts: int = None,
+        **options,
+    ):
+        """
+        Wrapper class for xESMF regridder class.
+
+        Available options: ``method``, ``periodic``, ``extrap_method``, ``extrap_dist_exponent``, ``extrap_num_src_pnts``
+
+        Parameters
+        ----------
+        input_grid : xr.Dataset
+            Contains source grid coordinates.
+        output_grid : xr.Dataset
+            Contains desintation grid coordinates.
+        method : str
+            Regridding method. Options are
+               - bilinear
+               - conservative
+               - conservative_normed
+               - patch
+               - nearest_s2d
+               - nearest_d2s
+        periodic : bool
+            Treat longitude as periodic. Used for global grids.
+        extrap_method : str
+            Extrapolation method. Options are
+               - inverse_dist
+               - nearest_s2d
+        extrap_dist_exponent : float
+            The exponent to raise the distance to when calculating weights for the extrapolation method.
+        extrap_num_src_pnts : int
+            The number of source points to use for the extrapolation methods that use more than one source point.
+
+        Raises
+        ------
+        KeyError
+            If data variable does not exist in the Dataset.
+        ValueError
+            If ``method`` is not valid.
+        ValueError
+            If ``extrap_method`` is not valid.
+
+        Examples
+        --------
+        Import xCDAT:
+
+        >>> import xcdat
+        >>> from xcdat.regridder import xesmf
+
+        Open a dataset:
+
+        >>> ds = xcdat.open_dataset("ts.nc")
+
+        Create output grid:
+
+        >>> output_grid = xcdat.create_gaussian_grid(32)
+
+        Create regridder:
+
+        >>> regridder = xesmf.XESMFRegridder(ds, output_grid, method="bilinear")
+
+        Regrid data:
+
+        >>> data_new_grid = regridder.horizontal("ts", ds, periodic=True)
+        """
+        super().__init__(input_grid, output_grid, **options)
+
+        if method not in VALID_METHODS:
+            raise ValueError(
+                f"{method!r} is not valid, possible options: {', '.join(VALID_METHODS)}"
+            )
+
+        if extrap_method is not None and extrap_method not in VALID_EXTRAP_METHODS:
+            raise ValueError(
+                f"{extrap_method!r} is not valid, possible options: {', '.join(VALID_EXTRAP_METHODS)}"
+            )
+
+        self._regridder = xe.Regridder(
+            self._input_grid,
+            self._output_grid,
+            method,
+            periodic=periodic,
+            extrap_method=extrap_method,
+            extrap_dist_exponent=extrap_dist_exponent,
+            extrap_num_src_pnts=extrap_num_src_pnts,
+        )
+
+    def horizontal(self, data_var: str, ds: xr.Dataset) -> xr.Dataset:
+        """
+        Regrid ``data_var`` in ``ds`` to output grid.
```

Comment:
```suggestion
        """Regrid ``data_var`` in ``ds`` to output grid.
```

Add description in docstring on how the algorithm/logic works